### PR TITLE
Make documentation of DataArray.where clearer

### LINKED ISSUE #7767: Inconsistency between xr.where() and da.where()
### What is your issue?

`xr.where()` and `da.where()` behave in seemingly opposite ways.

Example:

```python
da = xr.DataArray(np.arange(10)
print(xr.where(da < 5, 0, da).values)
print(da.where(da < 5, 0).values)
```
```
[0 0 0 0 0 5 6 7 8 9]
[0 1 2 3 4 0 0 0 0 0]
```

It seems like these two methods with the same name _should_ have the same functionality, but they give inverse results.

### PR DESCRIPTION
**Pull request by Riulinchen**
I tried to fix issue #7767, by adding a sentence to the Documentation of DataArray.where, as there seemed to be the largest confusion, whereas the documentation of xarray.where seems clear to me.

- [ ] Closes #7767

### CODE REVIEW

File: xarray/core/common.py [hunk 1]
```diff
@@ -1029,6 +1029,9 @@ def where(
         self: T_DataWithCoords, cond: Any, other: Any = dtypes.NA, drop: bool = False
     ) -> T_DataWithCoords:
         """Filter elements from this object according to a condition.
+        
+        Returns elements from 'DataArray', where 'cond' is True, 
+        otherwise fill in 'other' or 'drop'.
```
Comment (headtr1ck) [inline xarray/core/common.py:1034]: Maybe better to write "replace the rest by 'other' (default: NaN)"?

'or drop" is extremely vague, you could understand it as an alternative to 'other'. Not sure to what extent you should explain what drop does here vs. in the argument list.
Could be something like "drop completely masked elements along all dimensions", even though we don't talk about masking here... Maybe someone finds a better wording...

**Review by headtr1ck — COMMENTED** [review]
Thanks for the PR, always good to improve docs!
I think the current version is still a bit vague and does not add much benefit.

File: xarray/core/common.py (hunk 1 above)
Comment (Riulinchen) [reply xarray/core/common.py:1034]: What do you think about this?

Returns elements from 'DataArray', where 'cond' is True. The rest is replaced by 'other' (default: NaN) or dropped if drop=True.

Comment (keewis) [reply xarray/core/common.py:1034]: This sounds good, except we need to elaborate a bit on `drop`: A common confusion when using `where(..., drop=True)` with multi-dimensional arrays is that `cond` has to be all-`False` along any dimension for anything to be dropped, so we should definitely mentions that in the parameter description (have a look at the examples in the docstring if you're wondering what I'm talking about)

Comment (max-sixty) [reply xarray/core/common.py:1034]: > Returns elements from 'DataArray', where 'cond' is True. The rest is replaced by 'other' (default: NaN) or dropped if drop=True."

FWIW I think @Riulinchen 's proposal is much better than the current description, and we could merge that.

(agree we could also add more on how other dims are dropped, but don't need to gate the improvement on that?)

Comment (dcherian) [inline xarray/core/common.py:1034]:
```suggestion
        otherwise fill in 'other'.
```

Comment (keewis) [reply xarray/core/common.py]: sure, we can of course do that in a different PR (we just have to remember to actually do so)

**Pull request merged**